### PR TITLE
Removes empty partitions from given spill partition set

### PR DESCRIPTION
**Pull request by duanmeng**

### CODE REVIEW

**Comment by duanmeng** [comment]
@mbasmanova Hi Masha, I've created a separate PR according to your comments, could you please help take a look? Thanks. cc @xiaoxmeng

File: velox/exec/HashBuild.cpp [hunk 1]
```diff
@@ -709,17 +709,7 @@ bool HashBuild::finishHashBuild() {
...
-    // trigger unnecessary spilling at hash probe side.
-    auto iter = spillPartitions.begin();
-    while (iter != spillPartitions.end()) {
-      if (iter->second->numFiles() > 0) {
-        ++iter;
-      } else {
-        iter = spillPartitions.erase(iter);
-      }
-    }
+    removeEmptyPartitions(spillPartitions);
```
Comment (xiaoxmeng) [inline velox/exec/HashBuild.cpp:712]: Can we also use in RowNumber as well? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM. Thanks!

File: velox/exec/HashBuild.cpp (hunk 1 above)
Comment (duanmeng) [reply velox/exec/HashBuild.cpp:712]: Added.

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM. Thanks!

**Pull request closed without merging**